src/eko/couplings.py

```diff
@@ -206,23 +206,29 @@ def expanded_qcd(ref, order, nf, lmu):
         strong coupling at target scale :math:`a_s(\mu_R^2)`
     """
     res_as = ref
-    if order >= 1:
+    if order == 1:
```

Comment:
why did you change from `>=` to `==`? this way you have to repeat the definitions in all later branches

Reply:
I changed it because we were computing all the lower orders that we didn't need (e.g. for order==3 we compute as_LO, as_NLO and in the end as_NNLO)

Reply:
Well, then split the two things: maintain the original behavior, i.e. `<=`, to compute the beta coefficients, and introduce and second switch (a second block of `if ... elif ... else`) with the `==` you are proposing just for the solution :)

Reply:
In this way it will be as efficient as you obtained, but you don't need to duplicate the beta coefficients call

Reply:
Or you can just compute the coefficients always (they are very cheap) and only defer the solvers ...